File: reverse-integer/solve.ts
function reverse(x: number): number {
  let reversed: string = String(x).split("").reverse().join("").trim()
  let number = 0
  if (reversed[reversed.length - 1] === "-")
    number = Number(`-${reversed.slice(0, reversed.length - 1)}`)
  else number = Number(reversed)

  if (number < -(2 ** 31) || number > 2 ** 31 - 1) return 0
  return number
}

File: medium/longest-palindromic-substring/solve.go
package main

import (
	"strings"
)


func longestPalindrome(s string) string {
  if len(s) <= 1 {
		return s
	}

	longest := ""

	for i := 0; i < len(s); i++ {
		letters := string(rune(s[i]))

		if (len(longest) < 1) {
			longest += letters
		}

		for _, v := range s[i+1:] {
      letters += string(v)

      if (
        letters == strings.Join(reverse((strings.Split(letters, ""))), "") &&
        len(letters) > len(longest)) {
        longest = letters
			}
    }
	}

	return longest
}

func reverse(input []string) []string {
    if len(input) == 0 {
        return input
    }
    return append(reverse(input[1:]), input[0]) 
	}